Update SuperPoint model card

sbucaille opened the pull request:
# What does this PR do?

Updates SuperPoint model card for #36979 

## Before submitting
- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).

## Who can review?

@stevhliu

sbucaille committed:
docs: first draft to more standard SuperPoint documentation

stevhliu commented on docs/source/en/model_doc/superpoint.md:
```diff
@@ -10,48 +10,33 @@ specific language governing permissions and limitations under the License.
-description. The model is able to detect interest points that are repeatable under homographic transformations and
-provide a descriptor for each point. The use of the model in its own is limited, but it can be used as a feature
-extractor for other tasks such as homography estimation, image matching, etc.
+[SuperPoint](https://huggingface.co/papers/1712.07629) is the result of a self-supervised training of a fully-convolutional network for interest point detection and description. The model is able to detect interest points that are repeatable under homographic transformations and provide a descriptor for each point. The use of the model in its own is limited, but it can be used as a feature extrac ...
```
```suggestion
[SuperPoint](https://huggingface.co/papers/1712.07629) is the result of self-supervised training of a fully-convolutional network for interest point detection and description. The model is able to detect interest points that are repeatable under homographic transformations and provide a descriptor for each point. Usage on it's own is limited, but it can be used as a feature extractor for other tasks such as homography estimation and image matching.
```

stevhliu commented on docs/source/en/model_doc/superpoint.md:
```diff
@@ -10,48 +10,33 @@ specific language governing permissions and limitations under the License.
+[SuperPoint](https://huggingface.co/papers/1712.07629) is the result of a self-supervised training of a fully-convolutional network for interest point detection and description. The model is able to detect interest points that are repeatable under homographic transformations and provide a descriptor for each point. The use of the model in its own is limited, but it can be used as a feature extrac ...
 
-The abstract from the paper is the following:
+You can find all the original SuperPoint checkpoints under the [SuperPoint](https://huggingface.co/magic-leap-community/superpoint) repository.
```
```suggestion
You can find all the original SuperPoint checkpoints under the [Magic Leap Community](https://huggingface.co/magic-leap-community) organization.
```

stevhliu commented on docs/source/en/model_doc/superpoint.md:
```diff
@@ -10,48 +10,33 @@ specific language governing permissions and limitations under the License.
-to repeatedly detect a much richer set of interest points than the initial pre-adapted deep model and any other
-traditional corner detector. The final system gives rise to state-of-the-art homography estimation results on HPatches
-when compared to LIFT, SIFT and ORB.*
+> [!TIP]
+> Click on the SuperPoint models in the right sidebar for more examples of how to apply SuperPoint to different computer vision tasks.
```
```suggestion
> [!TIP]
> This model was contributed by [stevenbucaille](https://huggingface.co/stevenbucaille).
>
> Click on the SuperPoint models in the right sidebar for more examples of how to apply SuperPoint to different computer vision tasks.
```

stevhliu commented on docs/source/en/model_doc/superpoint.md:
```diff
@@ -10,48 +10,33 @@ specific language governing permissions and limitations under the License.
 <img src="https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/transformers/model_doc/superpoint_architecture.png"
 alt="drawing" width="500"/>
 
 <small> SuperPoint overview. Taken from the <a href="https://huggingface.co/papers/1712.07629v4">original paper.</a> </small>
```
```suggestion
```

stevhliu commented on docs/source/en/model_doc/superpoint.md:
```diff
@@ -10,48 +10,33 @@ specific language governing permissions and limitations under the License.
+> [!TIP]
+> Click on the SuperPoint models in the right sidebar for more examples of how to apply SuperPoint to different computer vision tasks.
 
 <img src="https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/transformers/model_doc/superpoint_architecture.png"
```
Let's move the image right after the first paragraph

stevhliu commented on docs/source/en/model_doc/superpoint.md:
````diff
@@ -64,46 +49,59 @@ processor = AutoImageProcessor.from_pretrained("magic-leap-community/superpoint"
-```python
+- SuperPoint outputs a dynamic number of keypoints per image, which makes it suitable for tasks requiring variable-length feature representations.
+
+```py
````
Indent the code block

stevhliu commented on docs/source/en/model_doc/superpoint.md:
````diff
@@ -64,46 +49,59 @@ processor = AutoImageProcessor.from_pretrained("magic-leap-community/superpoint"
+- The model provides both keypoint coordinates and their corresponding descriptors (256-dimensional vectors) in a single forward pass.
+- For batch processing with multiple images, you need to use the mask attribute to retrieve the respective information for each image. You can use the `post_process_keypoint_detection` from the `SuperPointImageProcessor` to retrieve the each image information.
 
+```py
````
Also indent this code block

stevhliu commented on docs/source/en/model_doc/superpoint.md:
````diff
@@ -64,46 +49,59 @@ processor = AutoImageProcessor.from_pretrained("magic-leap-community/superpoint"
-You can then print the keypoints on the image of your choice to visualize the result:
-```python
+- You can then print the keypoints on the image of your choice to visualize the result:
+```pyq
````
Indent as well

```suggestion
```py
```

stevhliu commented on docs/source/en/model_doc/superpoint.md:
```diff
@@ -117,12 +115,18 @@ plt.savefig(f"output_image.png")
 
+## Resources
+
+The resources provided in the following sections consist of a list of official Hugging Face and community (indicated by 🌎) resources to help you get started with SuperPoint. If you're interested in submitting a resource to be included here, please feel free to open a Pull Request and we'll review it! The resource should ideally demonstrate something new instead of duplicating an existing resource ...
```
Replace all the resources with only external links:

```md
Refer to this [noteboook](https://github.com/NielsRogge/Transformers-Tutorials/blob/master/SuperPoint/Inference_with_SuperPoint_to_detect_interest_points_in_an_image.ipynb) for an inference and visualization example.
```

stevhliu commented on docs/source/en/model_doc/superpoint.md:
```diff
@@ -117,12 +115,18 @@ plt.savefig(f"output_image.png")
 The original code can be found [here](https://github.com/magicleap/SuperPointPretrainedNetwork).
 
-## Resources
+<PipelineTag pipeline="keypoint-detection"/>
+
+- [`SuperPointForKeypointDetection`] is supported by this [example script](https://github.com/huggingface/transformers/tree/main/examples/pytorch/vision) and [notebook](https://colab.research.google.com/github/huggingface/notebooks/blob/main/examples/keypoint_detection.ipynb).
+- Check the [Keypoint detection task guide](../tasks/keypoint_detection) on how to use the model.
```
Can remove all of this

stevhliu commented on docs/source/en/model_doc/superpoint.md:
```diff
@@ -117,12 +115,18 @@ plt.savefig(f"output_image.png")
+- Check the [Keypoint detection task guide](../tasks/keypoint_detection) on how to use the model.
 
-A list of official Hugging Face and community (indicated by 🌎) resources to help you get started with SuperPoint. If you're interested in submitting a resource to be included here, please feel free to open a Pull Request and we'll review it! The resource should ideally demonstrate something new instead of duplicating an existing resource.
+**Visualization and inference**
```
```suggestion
```

stevhliu commented on docs/source/en/model_doc/superpoint.md:
```diff
@@ -132,13 +136,12 @@ A list of official Hugging Face and community (indicated by 🌎) resources to h
 
 ## SuperPointImageProcessor
 
-[[autodoc]] SuperPointImageProcessor
```
No need to change any of these `[[autodoc]]` classes here

stevhliu reviewed (commented):
Thanks for updating the model you contributed! ❤️

stevhliu commented:
> how is it possible to see the formatted result ?

I can run the `build_pr_documentation` job to generate a preview

sbucaille committed:
Apply suggestions from code review

Co-authored-by: Steven Liu <59462357+stevhliu@users.noreply.github.com>

sbucaille committed:
docs: reverted changes on Auto classes

sbucaille committed:
docs: addressed the rest of the comments

sbucaille commented:
@stevhliu addressed your comments, although I kept the line
```
- Check the [Keypoint detection task guide](../tasks/keypoint_detection) on how to use the model.
```
since the task exist I think it should be referred to somewhere even if there is no pipeline ?

stevhliu commented:
> since the task exist I think it should be referred to somewhere even if there is no pipeline no ?

I don't think the task guide adds any additional information that your model card doesn't already include so it seems a bit redundant to link to that guide.

HuggingFaceDocBuilderDev commented:
The docs for this PR live [here](https://moon-ci-docs.huggingface.co/docs/transformers/pr_38896). All of your documentation changes will be reflected on that endpoint. The docs are available until 30 days after the last update.

sbucaille committed:
docs: remove outdated reference to keypoint detection task guide in SuperPoint documentation

stevhliu committed:
Update superpoint.md

stevhliu reviewed (approved):
Thanks for your contribution!
